### ISSUE TITLE
Error during NAV App Build Creation with translation target file included

### ISSUE THREAD

**Issue by robertoscaringella** [Microsoft/AL#3528]
**Hi,
I'm developing an app with VS Code, with the parameter "features": "TranslationFile" in app.json.

I upload/Publish/install the app on NAV18CU5 WebClient without problem.

I created the file about the translation .it-IT.xlf and I Added the Target

<file datatype="xml" source-language="en-US" target-language="it-IT"original="AppName">

when I build the app (CTRL+SHIFT+B or CTRL+SHIFT+F5) the Visual Studio Code show this message on the output:**

_[Error - 3:33:06 PM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'al/createPackage' failed with error: 'Il valore non può essere null.
Nome parametro: value'
StackTrace:
   in Microsoft.CodeAnalysis.CSharp.ObjectDisplay.FormatLiteral(String value, ObjectDisplayOptions options)
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.CodeGenerator.GetLabelDataTypeFieldInitializer(TypeSymbol typeSymbol) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\CodeGenerator_Defaults.cs:riga 274
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.CodeGenerator.EmitFieldInitializer(TypeSymbol typeSymbol) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\CodeGen\CodeGenerator_Defaults.cs:riga 115
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.ApplicationObjectEmitHelper.EmitGlobalFieldDeclarations(ImmutableArray`1 globalVariables) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\Code\ApplicationObjectEmitHelper.cs:riga 773
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.ApplicationObjectEmitHelper.EmitGlobalVariables(ImmutableArray`1 globalVariables, SyntaxKind onClearAccessModifier, ClassDeclarationSyntax classDeclaration) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\Code\ApplicationObjectEmitHelper.cs:riga 139
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.ObjectCodeEmitter.Emit() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\Code\ObjectCodeEmitter.cs:riga 30
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.ApplicationObjectCodeEmitter.Emit() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\Code\ApplicationObjectCodeEmitter.cs:riga 30
   in Microsoft.Dynamics.Nav.CodeAnalysis.Emit.ApplicationObjectEmitter.EmitObjectCode() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Emitter\Code\ApplicationObjectEmitter.cs:riga 39
   in Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.EmitObjectCode(ObjectTypeSymbol containingType) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:riga 267
   in Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileObject(ObjectTypeSymbol containingType) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:riga 239
   in Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.<>c__DisplayClass17_0.<CompileObjectAsTask>b__0() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:riga 216
   in Microsoft.Dynamics.Nav.CodeAnalysis.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Utilities\UICultureUtilities.cs:riga 152
   in System.Threading.Tasks.Task.Execute()
--- Fine traccia dello stack da posizione precedente dove è stata generata l'eccezione ---
   in System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   in System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   in Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.WaitForWorkers() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:riga 161
   in Microsoft.Dynamics.Nav.CodeAnalysis.MethodCompiler.CompileMethodBodies(Compilation compilation, ModuleBuilder moduleBeingBuiltOpt, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compiler\MethodCompiler.cs:riga 145
   in Microsoft.Dynamics.Nav.CodeAnalysis.Compilation.CompileImpl(ModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\Compilation.cs:riga 810
   in Microsoft.Dynamics.Nav.CodeAnalysis.Compilation.Compile(ModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\Compilation.cs:riga 787
   in Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.PackageTheModule(Compilation compilation, CompilerTextWriter consoleOutput, CancellationToken cancellationToken, ErrorLogger errorLogger, AnalyzerDriver analyzerDriver) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:riga 531
   in Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.RunCore(CompilerTextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:riga 413
   in Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.Run(CompilerTextWriter consoleOutput, CancellationToken cancellationToken) in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:riga 312
   in Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.Extensions.CreatePackageRequestHandler.<HandleAsync>d__3.MoveNext() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\Extensions\CreatePackageRequestHandler.cs:riga 32
--- Fine traccia dello stack da posizione precedente dove è stata generata l'eccezione ---
   in System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   in System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   in Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.<Process>d__6.MoveNext() in C:\a\36\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:riga 59_

**If I create a simple app with one table and one page, and I do the same, the system work fine.

Could you help me?

Thank you,
Roberto.**

**Comment by kalberes** [comment]
does this repro on the newest docker image? If not  NAv 2018 related issues should "unfortunately" be handled using your regular support channel?

**Comment by kalberes** [comment]
I am confused a bit. What I can see here is that your publishing failed. This is not about symbol download. Isn't it?
Have you tried the same on the latest docker release? 
What I can see from this stack trace that one of labels in your app triggers an issue. 
Since I do not have proper repro steps, I can just guess.

**Comment by robertoscaringella** [comment]
Hi,
I found the reason of the problem, and the solution. 

Repro Steps:
- I added in app.json "features": ["TranslationFile"] and VS Code create the nameapp.g.xlf with the source Language (ENU). 
- I generated the file in language (for me Italian) copying the original file and I renamed in nameapp.it-IT.xlf.
- I filled **target-Language="it-IT"** here: <file datatype="xml" source-language="en-US" target-language="it-IT" original="nameapp">
- I tried to translate some string (not all), and I adding these translations in some target
- I built the app.

After many attempts I understand that VS Code return the errors bacause:
- It want all the target node defined
- All the target node must have a translation

My mistake was not create all the target node.

Now the app run.

Thank you.
Roberto.
